local/owr_local_media_source.c

```diff
@@ -548,6 +595,11 @@ static GstElement *owr_local_media_source_request_source(OwrMediaSource *media_s
                 goto done;
             }
 
+            CREATE_ELEMENT(priv->source_volume, "volume", "audio-source-volume");
+            g_object_set(priv->source_volume, "volume", priv->tmp_volume, NULL);
+            source_process = priv->source_volume;
+            gst_bin_add(GST_BIN(source_pipeline), source_process);
```

Comment:
I think it might make sense to see if the source element itself supports the `GstStreamVolume` interface, and use that as the volume element if it does. That would play nicer, for example, with sources that also support AGC and such.

Example code for this sort of thing: https://git.gnome.org/browse/rhythmbox/tree/backends/gstreamer/rb-player-gst.c#n336